modules/oe_paragraphs_timeline/oe_paragraphs_timeline.module

```diff
@@ -23,6 +24,95 @@ function oe_paragraphs_timeline_field_widget_timeline_widget_form_alter(&$elemen
     return;
   }
 
-  $element['label']['#required'] = TRUE;
-  $element['title']['#required'] = TRUE;
+  // We can't use the protected isDefaultValueWidget() method.
+  $is_default_value_widget = (bool) $form_state->get('default_value_widget');
+  if ($is_default_value_widget) {
+    return;
+  }
+
+  $element['label']['#required'] = $element['#required'];
+  $element['title']['#required'] = $element['#required'];
+  $element['#element_validate'][] = '_oe_paragraphs_validate_required_fields';
+
+  // If the elements were set as required, there is no need to append states.
+  if ($element['#required']) {
+    return;
+  }
+
+  // Get the selector to build the paths for each field.
+  $field_name = $field_definition->getName();
+  $delta = $element['#delta'];
+  $parents = $element['#field_parents'];
+  $parents[] = $field_name;
+  $parents[] = $delta;
```

Comment:
Now even don't need to declare `$field_name` and `$delta` variables.
It is possible to set like this:
  $parents[] = $field_definition->getName();
  $parents[] = $element['#delta'];

(it is very minor comment)